### PR TITLE
Add support for mimic joints.

### PR DESCRIPTION
**Pull request by destogl**
This PR adds the same support for mimic joints as in ﻿ros-simulation/gazebo_ros2_control#107.

The code is just slightly adjusted to work with Ignition.

@ahcorde: I have a question regarding writing position commands to the simulator from the plugin. It is a bit different from for velocity and effort. Is this wrong? I have adjusted this in [8298604](https://github.com/ignitionrobotics/ign_ros2_control/pull/33/commits/82986048051db72ae2a2b50cb6ced69031fb2d0b).

Moreover, I have a compilation issue with this code, but there is no error output about it. Do you have any idea?

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by ahcorde** [comment]
Ey @destogl,

I want to get rid of `JointPositionReset`, this is not the right way to use this ignition::gazebo::component. I would like to use this solution https://github.com/ignitionrobotics/ign_ros2_control/pull/29

**Comment by destogl** [comment]
> Ey @destogl,
> 
> I want to get rid of `JointPositionReset`, this is not the right way to use this ignition::gazebo::component. I would like to use this solution #29

Thanks! I will rebase this to the feature branch and update my code accordingly in the next few days.

**Comment by ahcorde** [comment]
friendly ping @destogl

**Comment by destogl** [comment]
Sorry, didn't manage to go back to it. Do you have any bandwidth to finish this. I will need few more days until I manage to come to it.

**Comment by ahcorde** [comment]
No worries, it was just a reminder

**Comment by livanov93** [comment]
@ahcorde Friendly ping for a review.

File: ign_ros2_control/src/ign_system.cpp [hunk 1]
```diff
@@ -200,6 +210,39 @@ bool IgnitionSystem::initSim(
...
+      } else {
+        mimic_joint.multiplier = 1.0;
+      }
+      RCLCPP_INFO_STREAM(
+        this->nh_->get_logger(),
+        "Joint '" << joint_name << "'is mimicing joint '" << mimicked_joint << "' with mutiplier: "
+                  << mimic_joint.multiplier);
+      this->dataPtr->mimic_joints_.push_back(mimic_joint);
+        suffix = "_mimic";
+
```
Comment (destogl) [inline ign_ros2_control/src/ign_system.cpp:243]:
```suggestion
```

Comment (livanov93) [reply ign_ros2_control/src/ign_system.cpp:264]: [bb465b3](https://github.com/ignitionrobotics/ign_ros2_control/pull/33/commits/bb465b394b0be207bd5fd6a5a07e0922668e13af)

**Review by ahcorde — CHANGES_REQUESTED** [review]
@destogl conflicts

**Comment by livanov93** [comment]
@destogl please change base branch to `master`

**Review by ahcorde — CHANGES_REQUESTED** [review]
there are some linter errors

can you merge the PR with `master`?

**Review by ahcorde — APPROVED** [review]

**Pull request merged**